### PR TITLE
feat: introduce PROCTIME()

### PR DESCRIPTION
**Pull request by yuhao-su**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- Introduce processing time. Users can specify a processing time on SOURCE/TABLE now.
- add a task local context for expression.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
~- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).~
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [ ] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

- SQL commands, functions, and operators

We can define a processing time on TABLE/SOURCE, which reflects the time a record is inserted in RisingWave.

```
CREATE TABLE t1 (proc_time Timestamptz as PROCTIME()) ;
```

### Release note

</details>

### CODE REVIEW

File: src/expr/src/expr/expr_proc_time.rs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    }
+}
+
+#[async_trait::async_trait]
+impl Expression for ProcTimeExpression {
+    fn return_type(&self) -> DataType {
+        DataType::Timestamptz
+    }
+
+    async fn eval(&self, input: &DataChunk) -> Result<ArrayRef> {
```
Comment (BugenZhao) [inline src/expr/src/expr/expr_proc_time.rs:56]: Since #9049 is merged, what about implementing `eval_v2` here?

**Review by BugenZhao — COMMENTED** [review]
Would you please add some tests? Rest LGTM.

**Review by BugenZhao — APPROVED** [review]

File: src/expr/src/expr/expr_proctime.rs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+impl Expression for ProcTimeExpression {
+    fn return_type(&self) -> DataType {
+        DataType::Timestamptz
+    }
+
+    async fn eval_v2(&self, input: &DataChunk) -> Result<ValueImpl> {
+        let proctime = CONTEXT
+            .try_with(|context| context.get_physical_time())
+            .map_err(|_| ExprError::Context)?;
+        let datum = Some(ScalarImpl::Int64(proctime as i64));
```
Comment (xiangjinwu) [inline src/expr/src/expr/expr_proctime.rs:58]: `Timestamptz` is in microseconds but seems `Epoch::physical_time` is in milliseconds?

Comment (BugenZhao) [reply src/expr/src/expr/expr_proctime.rs:58]: Good point. The interfaces of `Epoch` are ambiguous without including the time unit in the names.

Comment (yuhao-su) [reply src/expr/src/expr/expr_proctime.rs:58]: fix in #9126

**Pull request merged**